torch/csrc/jit/codegen/cuda/ir_iostream.cpp

```diff
@@ -241,12 +223,19 @@ void IrPrinter::handle(const Double* d) {
   }
 
   if (d->isSymbolic()) {
-    os_ << "d" << varName(d);
+    os_ << typePrefix(d->getDataType().value()) << varName(d);
   } else {
-    os_ << "double("
-        << std::setprecision(
-               std::numeric_limits<Double::ScalarType>::max_digits10)
-        << *(d->value()) << ")";
+    os_ << d->getDataType().value() << "(";
+    if (d->getDataType() == DataType::Double) {
+      os_ << std::setprecision(std::numeric_limits<double>::max_digits10)
+          << (float)*(d->value()) << ")";
+    } else if (d->getDataType() == DataType::Float) {
+      os_ << std::setprecision(std::numeric_limits<float>::max_digits10)
+          << *(d->value()) << ")";
```

Comment:
Should this be the following?
```suggestion
    if (d->getDataType() == DataType::Double) {
      os_ << std::setprecision(std::numeric_limits<double>::max_digits10)
          << *(d->value()) << ")";
    } else if (d->getDataType() == DataType::Float) {
      os_ << std::setprecision(std::numeric_limits<float>::max_digits10)
          << (float)*(d->value()) << ")";
```